crates/turborepo-lib/src/shim.rs

```diff
@@ -46,6 +46,24 @@ fn turbo_version_has_shim(version: &str) -> bool {
     version.major > 1
 }
 
+fn turbo_version_in_range(version: &str, root_path: &Path) -> bool {
+    let version = Version::parse(version).unwrap();
+
+    let version_req_string = match read_turbo_json(root_path) {
+        Ok(turbo_json) => turbo_json.turboVersion,
+
+        // Either not present or it failed to read `turbo.json`.
+        // We'll assume this is *.
+        Err(_) => "*".to_string(),
+    };
+
+    // FIXME: this SemVer matching is "Rust" SemVer matching.
+    match VersionReq::parse(version_req_string.as_str()) {
+        Ok(version_request) => version_request.matches(&version),
+        Err(_) => false,
+    }
```

Comment:
For this I think we'll need to add a `use anyhow::Context` import statement
```suggestion
    let version_req = VersionReq::parse(version_req_string).context("Invalid semver range in turbo.json")?;
    if !version_req.matches(&version) {
        return Err(anyhow!("Project specifies that version is {}, you're running {}.", version_req, version));
    }
    Ok(())
```